sdk/storage/azblob/zt_examples_test.go

```diff
@@ -52,76 +37,72 @@ func Example() {
 	if !ok {
 		panic("AZURE_STORAGE_PRIMARY_ACCOUNT_KEY could not be found")
 	}
-	cred, err := NewSharedKeyCredential(accountName, accountKey)
+	cred, err := azblob.NewSharedKeyCredential(accountName, accountKey)
 	if err != nil {
 		log.Fatal(err)
 	}
 
-	// Open up a service client.
-	// You'll need to specify a service URL, which for blob endpoints usually makes up the syntax http(s)://<account>.blob.core.windows.net/
-	service, err := NewServiceClientWithSharedKey("https://"+accountName+".blob.core.windows.net/", cred, nil)
+	// The service URL for blobs endpoints is usually in the form: http(s)://<account>.blob.core.windows.net/
+	service, err := azblob.NewServiceClientWithSharedKey(fmt.Sprintf("https://%s.blob.core.windows.net/", accountName), cred, nil)
 	if err != nil {
 		log.Fatal(err)
 	}
 
-	// All operations in the Azure Storage Blob SDK for Go operate on a context.Context, allowing you to control cancellation/timeout.
-	ctx := context.Background() // This example has no expiry.
-
-	// This example showcases several common operations to help you get started, such as:
-
 	// ===== 1. Creating a container =====
 
-	// First, branch off of the service client and create a container client.
+	// First, create a container client, and use the Create method to create a new container in your account
 	container := service.NewContainerClient("myContainer")
-	// Then, fire off a create operation on the container client.
-	// Note that, all service-side requests have an options bag attached, allowing you to specify things like metadata, public access types, etc.
-	// Specifying nil omits all options.
-	_, err = container.Create(ctx, nil)
+	// All functions that make service requests have an options struct as the final parameter.
+	// The options struct allows you to specify optional parameters such as metadata, public access types, etc.
+	// If you want to use the default options, pass in nil.
+	_, err = container.Create(context.TODO(), nil)
 	if err != nil {
 		log.Fatal(err)
 	}
 
 	// ===== 2. Uploading/downloading a block blob =====
-	// We'll specify our data up-front, rather than reading a file for simplicity's sake.
 	data := "Hello world!"
 
-	// Branch off of the container into a block blob client
+	// Create a new BlockBlobClient from the ContainerClient
 	blockBlob := container.NewBlockBlobClient("HelloWorld.txt")
 
 	// Upload data to the block blob
-	_, err = blockBlob.Upload(ctx, NopCloser(strings.NewReader(data)), nil)
+	_, err = blockBlob.Upload(context.TODO(), streaming.NopCloser(strings.NewReader(data)), nil)
```

Comment:
Just curious why we started using `context.TODO()` instead of `context.Background()`?

Reply:
Another thing to file an issue for if we haven't already: modifying the Upload function to take a ReadSeeker instead of a ReadSeekCloser, removing the need to use NopCloser.

Reply:
I use `context.TODO()` in documentation for two reasons: 
1. It's easily findable with a search of "TODO" in a code editor and it grabs the eye of something to change. TODO is also equal to Background in 
2. Also per the [documentation](https://pkg.go.dev/context#TODO) it's recommended
> TODO returns a non-nil, empty Context. Code should use context.TODO when it's unclear which Context to use or it is not yet available (because the surrounding function has not yet been extended to accept a Context parameter).

Reply:
Should we then change the other examples? I saw some later in this file that still use Background.

Reply:
Yes, thanks for catching that. Updated.